brickstrap-components.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+#
+# This file is part of brickstrap.
+#
+# brickstrap - create a foreign architecture rootfs using kernel namespaces,
+#              multistrap, and qemu usermode emulation and create a disk image
+#              using libguestfs
+#
+# Copyright (C) 2016 Johan Ouwerkerk <jm.ouwerkerk@gmail.com>
+#
+
+#
+# Note: this file is not meant to be executed, source it as a library of
+# functions instead. Variables used by the functions (other than stack) are
+# namespaced using the 'BR_' or 'BRP_' prefixes. Function names are namespaced
+# similarly, using the 'br_' and 'brp_' prefixes. See docs/namespacing.md for
+# more information on namespacing in brickstrap.
+#
+
+function brp_validate_project_name()
+{
+    [ -n "$BR_PROJECT" ] || fail "Project name must not be empty"
+    if [ -r "$BR_PROJECT" -a -d "$BR_PROJECT" ]; then
+        BR_PROJECT_DIR=$(readlink -f "$BR_PROJECT")
+    elif [ -r "$(br_script_path)/$BR_PROJECT" ] && \
+        [ -d "$(br_script_path)/$BR_PROJECT" ]; then
+        BR_PROJECT_DIR=$(readlink -f "$(br_script_path)/$BR_PROJECT")
+    else
+        fail "Invalid project name: '$BR_PROJECT'"
+    fi
+}
+
+function br_project_dir()
+{
+    [ -n "$BR_PROJECT_DIR" ] && echo -n "$BR_PROJECT_DIR"
+}
+
+function brp_validate_component_name()
+{
+    [ $# -eq 1 -a -n "$1" ] || fail "Component name must not be empty"
+    [ -d "$BR_PROJECT_DIR/$1" ] || fail "Not a valid component name: '$1'"
```

Comment:
I would find it useful for the error message to say why it is not valid so that I know how to fix it.